kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/dsl/ConfigMapResource.java

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.dsl;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+
+public interface ConfigMapResource extends Resource<ConfigMap>,
```

Comment:
As a method on ConfigMapResource you can do something like configMaps().resource(item).fromFile, or configMaps().load(stream).fromFile.  This could be put at a higher level, like configMaps().fromFile(name, path), to eliminate a bit of overlap between the context item and what's coming from the file.

Reply:
I moved it a level up by making `client.configMaps()` return `FromFileCreatableOperation` but now it's not able to use  client like this `client.configMaps().inNamespace("foo").fromFile(..)`

Do you think we should consider this scenario? There is also an option to add an overloaded `fromFile` method that receives `ObjectMeta` instead of `name`

Reply:
> but now it's not able to use client like this client.configMaps().inNamespace("foo").fromFile(..)

To support that either MixedOperation would need to be generic to allow for a refined NonNamespaceOperation type, or there's NamespaceableResource - that allow a Resource to have a namespace specified when the root operation does not have one.

> There is also an option to add an overloaded fromFile method that receives ObjectMeta instead of name

Makes sense, that would make this more generalized than what kubectl supports.

Reply:
Considering both semantics and consistency, I'd expect that `client.resource` and `client.configMaps().fromFile` did exactly the same. 

So IMHO (and just with the context from reading this thread) I'd expect;
`client.configMaps().fromFile($theFile).inNamespace("default").operation()`